aws/src/main/java/org/apache/iceberg/aws/s3/S3URI.java

```diff
@@ -50,7 +52,24 @@
    * @param location fully qualified URI
    */
   S3URI(String location) {
+    this(location, ImmutableMap.of());
+  }
+
+  /**
+   * Creates a new S3URI in the form of scheme://(bucket|accessPoint)/key?query#fragment with additional information
+   * on accessPoints.
+   * <p>
+   * The URI supports any valid URI schemes to be backwards compatible with s3a and s3n,
+   * and also allows users to use S3FileIO with other S3-compatible object storage services like GCS.
+   * If the bucket in the location has an access point in the mapping, the access point is used to perform all the S3
+   * operations
+   *
+   * @param location fully qualified URI
+   * @param bucketToAccessPointMapping contains mapping of bucket to access point
+   */
+  S3URI(String location, Map<String, String> bucketToAccessPointMapping) {
     Preconditions.checkNotNull(location, "Location cannot be null.");
+    Preconditions.checkNotNull(bucketToAccessPointMapping, "BucketToAccessPointMapping cannot be null.");
```

Comment:
This error message probably doesn't make sense to people that will see it. We avoid using variable or class names in error messages because users aren't familiar with Iceberg code internal details.

In addition, I don't think there's anything the user would be able to do about this because the mapping is created by AWS code. Instead, I think this should just default the mapping to an empty immutable map.